kafka: add production docs (#101)
kudobuilder/frameworks: repository/kafka/docs/latest/concepts.md

![kudo-kafka](./resources/images/kudo-kafka.png)

# KUDO Kafka Concepts

> KUDO Kafka is a Kubernetes operator build on top of [KUDO](kudo.dev) and requires KUDO

Reviewer: ```suggestion
KUDO Kafka is a Kubernetes operator built on top of [KUDO](kudo.dev) and requires KUDO
```

#### KUDO Kafka CRDs

> There are three CRDs, that are installed for KUDO Kafka.

Reviewer: ```suggestion
There are three CRDs that are installed when deploying KUDO Kafka:
```

- Operator
- OperatorVersion
- Instance

> An Instance represents an instantiation of a KUDO Kafka cluster. The Operator and OperatorVersion hold all knowledge to instantiate the KUDO Kafka cluster. 

Reviewer: Should we combine these descriptions with the list above? I mean:
```
- Operator: the definition that describes the Kudo Kafka operator.
- OperatorVersion: the definition that describes the Kudo Kafka operator for a specific version.
- Instance: the instantiation of a KUDO Kafka cluster based on the OperatorVersion.

#### KUDO Controller Reconcile Cycle

> KUDO controller watches the CRDs of Operator, OperatorVersion and Instance in Kubernetes API

Reviewer: ```suggestion
The KUDO controller continually watches the Operator, OperatorVersion and Instance CRDs via the Kubernetes API.
```

![kudo-kafka](./resources/images/kudo-controller-kafka.png)

> When the user install the KUDO Kafka using kudo-cli, it creates the KUDO Kafka CRDs for Operator, OperatorVersion and Instance. More informration can be read in [KUDO Architecture](https://kudo.dev/docs/architecture.html#architecture-diagram) 

Reviewer: ```suggestion
When a user installs KUDO Kafka using the `kudo-cli`, the controller creates the KUDO Kafka CRDs for Operator, OperatorVersion and Instance. More information can be read in [KUDO Architecture](https://kudo.dev/docs/architecture.html#architecture-diagram) 
```

![kudo-kafka](./resources/images/kudo-installs-kafka.png)

> When KUDO Controller detects a new `Instance` it creates all the resources required to reach the desired state of the configuration. 

Reviewer: ```suggestion
When the KUDO Controller detects a new `Instance`, it creates all the resources required to reach the desired state of the configuration. 
```

![kudo-kafka](./resources/images/kafka-cluster.png)

> Same process is followed for any updates and deletion that is handled by the KUDO Controller.

Reviewer: ```suggestion
The same process is followed for any updates or deletions. Everything is handled by the KUDO Controller.
```
